tests/PHPStan/Analyser/data/json-decode/narrow_type_with_force_array.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Analyser\JsonDecode;
+
+use function PHPStan\Testing\assertType;
+
+$value = json_decode('true', true);
+assertType('true', $value);
+
+$value = json_decode('1', true);
+assertType('1', $value);
+
+$value = json_decode('1.5', true);
+assertType('1.5', $value);
+
+$value = json_decode('false', true);
+assertType('false', $value);
+
+$value = json_decode('{}', true);
+assertType('array{}', $value);
+
+$value = json_decode('[1, 2, 3]', true);
+assertType('array{1, 2, 3}', $value);
```

Comment:
What's gonna happen in case of invalid json?

Reply:
Looks like `null` is returned: https://3v4l.org/oPLja

Reply:
I've added test fixture for it :+1: